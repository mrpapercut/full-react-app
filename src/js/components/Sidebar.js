import {Component, createElement as E} from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';

import createElement from '~/util/createElement';

const [div, h2] = ['div', 'h2'].map(createElement);

class Sidebar extends Component {
    constructor(props) {
        super(props);
    }

    getLinks() {
        return [{
            to: '/',
            componentName: 'MainComponent'
        }, {
            to: '/image',
            componentName: 'ImageComponent'
        }, {
            to: '/state',
            componentName: 'StateComponent'
        }, {
            to: '/store',
            componentName: 'StoreComponent'
        }];
    }

    render() {
        const links = this.getLinks();

        return div({
            className: 'sidebar'
        },
            h2({
                className: 'sidebar-title'
            }, 'Sidebar'),
            links.map(({to, componentName}) => {
                return div({
                    className: 'sidebar-item'
                },
                    E(Link, {
                        to,
                        key: `link-${componentName.toLowerCase()}`
                    }, componentName)
                )
            })
        );
    }
}

const mapStateToProps = ({mainReducer}) => ({

});

const mapDispatchToProps = dispatch => ({

});

export default connect(mapStateToProps, mapDispatchToProps)(Sidebar);
