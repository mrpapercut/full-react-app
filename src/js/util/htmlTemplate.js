const htmlTemplate = ({title, body, scriptTags, preloadedState}) => `<!doctype html>
<html>
<head>
    <meta charset="utf8"></meta>
    <title>${title}</title>
    <base href="/">

    <meta name="viewport" content="width=device-width, initial-scale=1.0">
</head>
<body>
    <div id="appwrapper">${body}</div>
    <script>${preloadedState}</script>
    ${scriptTags}
</body>
</html>
`;

export default htmlTemplate;
