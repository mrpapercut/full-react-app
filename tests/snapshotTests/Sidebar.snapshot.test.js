import {createElement} from 'react';
import renderer from 'react-test-renderer';

import {wrapElementWithRouter} from '../util/snapshot-test-util';

import Sidebar from '~/components/Sidebar';

test('Sidebar', () => {
    const component = renderer.create(wrapElementWithRouter(createElement(Sidebar)));
    expect(component.toJSON()).toMatchSnapshot();
});
