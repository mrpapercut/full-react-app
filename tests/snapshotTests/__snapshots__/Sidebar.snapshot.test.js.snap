// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar 1`] = `
<div
  className="sidebar"
>
  <h2
    className="sidebar-title"
  >
    Sidebar
  </h2>
  <div
    className="sidebar-item"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      MainComponent
    </a>
  </div>
  <div
    className="sidebar-item"
  >
    <a
      href="/image"
      onClick={[Function]}
    >
      ImageComponent
    </a>
  </div>
  <div
    className="sidebar-item"
  >
    <a
      href="/state"
      onClick={[Function]}
    >
      StateComponent
    </a>
  </div>
  <div
    className="sidebar-item"
  >
    <a
      href="/store"
      onClick={[Function]}
    >
      StoreComponent
    </a>
  </div>
</div>
`;
